draft-ietf-tls-tls13.md

```diff
@@ -1817,6 +1817,17 @@ the 0-RTT data without detection. If the client's ephemeral share
 is compromised and client authentication is used, then the attacker
 can impersonate the client on subsequent connections.
 
+It is up to the implementation and application layers above TLS to mitigate
+the impact of these weaknesses.  For example, to support strong client
+authentication and forward secrecy over 0-RTT connections, it is possible to
+issue a new ticket on the first 1-RTT connection, and on every resumed session
+after that.  Each co-located group of servers could ensure that tickets they
+issue are used only once by maintaining a local cache of ticket decryption
+keys.  An accurate clock on the client could ensure freshness of client
```

Comment:
I agree with your concerns, but at the same time, I feel the document would be improved with some guidance about mitigation.  As it stands, the warnings read like a Stay Away sign.  Another direction is to accept the forward secrecy limitations and implement QUIC-inspired strike registers, without using tickets.  This can give client certificates more value, and many people don't care about a few days of lost forward secrecy.

In any case, dealing with replay when implementing HTTPS involves making all requests idempotent at the application layer, even POST requests.  This is true for TLS 1.2 as well.  All requests are subject to being replayed, either due to network errors or because of a malicious attack.